datasets/wili_2018/wili_2018.py

```diff
@@ -0,0 +1,342 @@
+    "ita",
+    "kab",
+    "bel",
+    "ang",
+    "mhr",
+    "che",
+    "koi",
+    "glv",
+    "ido",
+    "fao",
+    "bak",
+    "isl",
+    "bcl",
+    "tet",
+    "jpn",
+    "kur",
+    "map-bms",
+    "tyv",
+    "olo",
+    "arg",
+    "ori",
+    "lim",
+    "tel",
+    "lin",
+    "roh",
+    "sqi",
+    "xho",
+    "mlg",
+    "fas",
+    "hbs",
+    "tam",
+    "aze",
+    "lad",
+    "nob",
+    "sin",
+    "gla",
+    "nap",
+    "snd",
+    "ast",
+    "mal",
+    "mdf",
+    "tsn",
+    "nds",
+    "tgl",
+    "nno",
+    "sun",
+    "lzh",
+    "jbo",
+    "crh",
+    "pap",
+    "oci",
+    "hak",
+    "uzb",
+    "zho",
+    "hsb",
+    "sme",
+    "mlt",
+    "vep",
+    "lez",
+    "nld",
+    "nds-nl",
+    "mrj",
+    "spa",
+    "ceb",
+    "ina",
+    "heb",
+    "hun",
+    "que",
+    "kaa",
+    "mar",
+    "vec",
+    "frp",
+    "ell",
+    "sah",
+    "eus",
+    "ces",
+    "slk",
+    "chr",
+    "lij",
+    "nep",
+    "srd",
+    "ilo",
+    "be-tarask",
+    "bod",
+    "orm",
+    "war",
+    "glg",
+    "mon",
+    "gle",
+    "min",
+    "ibo",
+    "ile",
+    "epo",
+    "lav",
+    "lrc",
+    "als",
+    "mzn",
+    "rup",
+    "fur",
+    "tat",
+    "myv",
+    "pan",
+    "ton",
+    "kom",
+    "wuu",
+    "tcy",
+    "tuk",
+    "kan",
+    "ltg",
+]
+
+
+class Wili2018(datasets.GeneratorBasedBuilder):
+    """WiLI Language Identification Dataset"""
+
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="WiLI-2018 dataset",
+            version=VERSION,
+            description="Plain text of import of WiLI-2018",
+        )
+    ]
+
+    def _info(self):
+
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=datasets.Features(
+                {"sentence": datasets.Value("string"), "label": datasets.features.ClassLabel(names=_CLASSES)}
+            ),
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        train_path = dl_manager.download_and_extract(_TRAIN_DOWNLOAD_URL)
+        test_path = dl_manager.download_and_extract(_TEST_DOWNLOAD_URL)
+        return [
+            datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"filepath": train_path}),
+            datasets.SplitGenerator(name=datasets.Split.TEST, gen_kwargs={"filepath": test_path}),
+        ]
+
+    def _generate_examples(self, filepath):
+
+        df = pd.read_csv(filepath, encoding="utf8", engine="python")
+        sentences = df.iloc[:, 0].tolist()
+        labels = df.iloc[:, 1].tolist()
+
+        ids = [i for i in range(len(labels))]
+
+        for id_, sentence, label in zip(ids, sentences, labels):
```

Comment:
Can you iterate over the lines of the csv file instead of using pandas ?
It will reduce the RAM usage significantly (hundreds of MB)

Reply:
And I tried to use CSV reader instead of pandas but wasn't able to succeed in it.
For example, a data observation looks like this (one observation in each line):
Hi, I am a human, 10
^                      ^^ ^
|...............................| |  |
   sentence          ,  label

So if I try separate sentences and labels with a delimiter ',' (and each line denotes one example), I wasn't able to do it as a sentence may include a ',' in it. @lhoestq Can you suggest something else, that I can try?